wpengine/faustjs
# RFC: Set 404 Revalidate timer to 900 seconds.

## Tasks


## Description

Sets the default ISR revalidate timer on 404 pages.

Closes https://github.com/wpengine/faustjs/issues/1580

## Related Issue(s):



## Testing

1. Checkout and Build the project.
2. Run the project in production mode (npm run start) vs in dev mode (npm run dev)
3. Run a curl command that will tigger a 404 response.
4. You should see the following header in cache-control:
` Cache-Control: s-maxage=900, stale-while-revalidate`

## Screenshots



**Before**
```
❯ curl -vvv http://localhost:3000/222 | grep cache-control
...
>
< HTTP/1.1 404 Not Found
< x-using: faust
< x-nextjs-cache: HIT
< Cache-Control: s-maxage=31536000, stale-while-revalidate
< ETag: "inm7ps8d27200"
< Content-Type: text/html; charset=utf-8
< Content-Length: 2592
< Vary: Accept-Encoding
< Date: Thu, 09 Nov 2023 12:28:23 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
```

**After**

```
❯ curl -vvv http://localhost:3000/222 | grep cache-control
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 127.0.0.1:3000...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /222 HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/8.1.2
> Accept: */*
>
< HTTP/1.1 404 Not Found
< x-using: faust
< x-nextjs-cache: HIT
< Cache-Control: s-maxage=900, stale-while-revalidate
< ETag: "7hstphb9w0200"
< Content-Type: text/html; charset=utf-8
< Content-Length: 2592
< Vary: Accept-Encoding
< Date: Thu, 09 Nov 2023 12:33:07 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
<
{ [2592 bytes data]
```

## Documentation Changes



## Dependant PRs
